arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+# Bucket management in soda-multicloud
+
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10), [Vineela Pachchipulusu]()
+
+
+## Motivation and background
+
+In the cloud world, every cloud vendor provides storage space to users. Generally in terms of bucket and folder. A user can store the data/objects in a specific bucket or folder.
+
+SODA multi-cloud(gelato) is a project which manages multiple cloud vendors from one place. And also interfaces to use cloud services. So it is important for soda to provide bucket management support for various cloud vendors. So that user can create and manage buckets from one place(soda multi-cloud).
+
+## Goals
+The document provides the use case, design and implementation guidelines for supporting bucket management from SODA platform. It includes creation, deletion, object uploading, migration etc. in various cloud providers.
+
+### Non-Goals
+NA
+
+### Requirement Analysis
+
+##### Functional Requirements
+
+1. Create buckets in various cloud providers. Ex: Aws, gcp, azure
+
+2. Delete bucket from various cloud providers. Ex: Aws, gcp, azure
+
+3. Upload objects in bucket
+
+4. Set lifecycles rules on objects
+
+5. Migrate objects from one location to different locations
+
+
+##### Non Functional Requirements
+
+1. The system should be highly available. Multiple users can login to the system and create buckets, folders, upload objects etc.
+The design document for this can be referred from [here](https://github.com/sodafoundation/architecture-analysis/pull/96)
+
+2. The system should work with minimum latency.
+
+3. The system should be scalable and it should handle the high amount of users request.
+
+
+## Architecture Analysis
+
+The overall multi cloud architecture can be referred from [here](https://github.com/sodafoundation/architecture-analysis/blob/master/arch-design/multicloud/MultiCloud_Design.md)
+
+### Use case View
```

Comment:
did not understand the use case here. Question is
- What is the advantage one gets by using SODA to manage the buckets, vs. managing them directly on each cloud account ? Is this just a same-functionality-aggregated-in-one-place feature ?

Reply:
Any advantage are we getting considering the user-space? One User can have buckets from different cloud vendors with the policies given to that particular user? Considering existing IAM service for multi-cloud, will be consider this because (IAM using AK/SK) at cloud vendor will be different from the IAM service at SODA multi-cloud Or Will be leverage the IAM from the cloud vendor itself?

Reply:
> did not understand the use case here. Question is
> 
> * What is the advantage one gets by using SODA to manage the buckets, vs. managing them directly on each cloud account ? Is this just a same-functionality-aggregated-in-one-place feature ?

Yes. The main use case is managing the all different cloud buckets from one place

Reply:
> Any advantage are we getting considering the user-space? One User can have buckets from different cloud vendors with the policies given to that particular user? Considering existing IAM service for multi-cloud, will be consider this because (IAM using AK/SK) at cloud vendor will be different from the IAM service at SODA multi-cloud Or Will be leverage the IAM from the cloud vendor itself?

Yes. One user can have multiple buckets from different cloud vendors. For now, IAM service of soda multi-cloud and cloud vendors are same.